crates/bevy_ecs/src/schedule/graph/graph_map.rs

```diff
@@ -249,28 +248,65 @@ where
     }
 
     /// Return an iterator over all edges of the graph with their weight in arbitrary order.
-    pub fn all_edges(&self) -> impl ExactSizeIterator<Item = (NodeId, NodeId)> + '_ {
-        self.edges.iter().copied().map(CompactNodeIdPair::load)
+    pub fn all_edges(&self) -> impl ExactSizeIterator<Item = (N, N)> + '_ {
+        self.edges.iter().copied().map(N::Pair::unwrap)
     }
 
-    pub(crate) fn to_index(&self, ix: NodeId) -> usize {
+    pub(crate) fn to_index(&self, ix: N) -> usize {
         self.nodes.get_index_of(&ix).unwrap()
     }
+
+    /// Converts from one [`GraphNodeId`] type to another. If the conversion fails,
+    /// it returns the error from the target type's [`TryFrom`] implementation.
+    ///
+    /// # Errors
+    ///
+    /// If the conversion fails, it returns an error of type `T::Error`.
+    pub fn try_into<T: GraphNodeId + TryFrom<N>>(self) -> Result<Graph<DIRECTED, T, S>, T::Error>
```

Comment:
Suggestions welcome on making this function a bit more legible (it works and I'm lazy).

Reply:
Would it help to reframe it as `map`?  Part of the complexity here is bubbling up the conversion errors just to unwrap them with `unreachable!`.  If you instead passed a closure, then you could put the panic in there and not deal with error handling here.  

Otherwise, my best suggestion is to do a little more type inference as `.collect::<Result<_, T::Error>>()?`.  Rust needs help on the error type because `?` does `into()`, but it can infer the success type.

Reply:
I've cleaned it up a bit with some inner functions, docs, and used the `_` as suggested. Should I take it further?